woocommerce/woocommerce
# [Accessibility] Keep focus on shipping option input once selected

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #37606  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. On the admin go to WooCommerce -> Settings -> Shipping and add at least three shipping methods.
2. Make sure the store is using the Classic Cart and Checkout page. 
3. On the store, add a product to the cart and go to the Cart page.
4. Focus on a shipping radio input in the Cart totals section.
5. Change the shipping method by pressing the arrow keys (up/down) on the keyboard.
6. Once the shipping method changes, verify the focus is still on the selected shipping method.
7. Repeat the steps on the shipping methods section on the Checkout page.
8. **HEADS-UP** – You may notice that VoiceOver is announcing the selected shipping method and then starts announcing it again. This is because once the request to change the shipping method returns a response, the whole Cart totals section is replaced with new HTML content, and the focus is moved to the selected shipping method again. It's as if the focus had changed to a different element with the same content. That's why VoiceOver starts announcing it again from the beginning.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
